src/react/account/AccountRoleSelectButtonAndModal.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState } from 'react';
+import { useRouteMatch } from 'react-router';
+import Table from '@scality/core-ui/dist/components/tablev2/Tablev2.component';
+import { Button } from '@scality/core-ui/dist/next';
+import { Tooltip } from '@scality/core-ui';
+import SpacedBox from '@scality/core-ui/dist/components/spacedbox/SpacedBox';
+import { CustomModal as Modal, ModalBody } from '../ui-elements/Modal';
+import { regexArn, SCALITY_INTERNAL_ROLES, useAccounts } from '../utils/hooks';
+import {
+  useCurrentAccount,
+  useDataServiceRole,
+} from '../DataServiceRoleProvider';
+import { getRoleArnStored, setRoleArnStored } from '../utils/localStorage';
+import { Icon } from '../ui-elements/Help';
+import { AccountSelectorButton } from '../ui-elements/Table';
+
+function AccountRoleSelectButtonAndModal() {
+  const accounts = useAccounts();
+  const { path } = useRouteMatch();
+  const { account, selectAccountAndRoleRedirectTo } = useCurrentAccount();
+  const accountName = account?.Name;
+  const { roleArn } = useDataServiceRole();
+  const [assumedRoleArn, setAssumedRoleArn] = useState(roleArn);
+  const [currentAccountName, setCurrentAccountName] = useState(accountName);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const accountsWithRoles = [];
+  accounts.forEach((account) => {
+    const accountName = account.Name;
+    account.Roles.forEach((role) => {
+      const rolePath = regexArn.exec(role.Arn)?.groups['role_path'] || '';
+      const roleName = regexArn.exec(role.Arn)?.groups['role_name'] || '';
```

Comment:
nit: regex could be executed only once as it can be a bit costly to execute
```
const parsedArn = regexArn.exec(role.Arn); 
const rolePath = parsedArn?.groups['role_path'] || '';
const roleName = parsedArn?.groups['role_name'] || '';
```

Reply:
Good point! Thanks for the comments!!